$OpenBSD: patch-transport_ml,v 1.1 2005/05/30 19:27:45 sturm Exp $
Post-release developer patch.
o Fix the assert failure on all platforms.
  See http://groups.yahoo.com/group/unison-users/message/3200
--- transport.ml.orig	Mon Sep  6 15:15:46 2004
+++ transport.ml	Sun May 22 16:14:32 2005
@@ -15,7 +15,8 @@ let fileSize uiFrom uiTo =
   match uiFrom, uiTo with
     _, Updates (File (props, ContentsUpdated (_, _, ress)), _) ->
       (Props.length props, Osx.ressLength ress)
-  | Updates (File _, Previous (_, props, _, ress)), NoUpdates ->
+  | Updates (_, Previous (`FILE, props, _, ress)),
+    (NoUpdates | Updates (File (_, ContentsSame), _)) ->
       (Props.length props, Osx.ressLength ress)
   | _ ->
       assert false
